File: examples/demo-app/src/summarize.ts
// Negative example: free-text output, should NOT be flagged.
import OpenAI from "openai";
import { zodResponseFormat } from "openai/helpers/zod";
import { z } from "zod";

const openai = new OpenAI();

export async function summarize(doc: string) {
  const res = await openai.chat.completions.parse({
    model: "gpt-5.6-terra",
    messages: [{ role: "user", content: `Summarize:\n${doc}` }],
    response_format: zodResponseFormat(z.object({ summary: z.string(), title: z.string() }), "summary"),
  });
  return res.choices[0].message.parsed;
}

File: examples/demo-app/py/summarize.py
# Negative example: free-text output, should NOT be flagged.
from openai import OpenAI
from pydantic import BaseModel

client = OpenAI()


class Summary(BaseModel):
    title: str
    body: str


def summarize(doc: str) -> Summary:
    return client.chat.completions.parse(
        model="gpt-5.6-terra",
        messages=[{"role": "user", "content": f"Summarize:\n{doc}"}],
        response_format=Summary,
    ).choices[0].message.parsed
